src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -20,35 +43,117 @@ internal TagHelperAttributeDescriptor(string name, PropertyInfo propertyInfo)
         /// Instantiates a new instance of the <see cref="TagHelperAttributeDescriptor"/> class.
         /// </summary>
         /// <param name="name">The HTML attribute name.</param>
-        /// <param name="propertyName">The name of the CLR property name that corresponds to the HTML
-        /// attribute.</param>
+        /// <param name="propertyName">The name of the CLR property that corresponds to the HTML attribute.</param>
         /// <param name="typeName">
-        /// The full name of the named (see <paramref name="propertyName"/>) property's
-        /// <see cref="System.Type"/>.
+        /// The full name of the named (see <paramref name="propertyName"/>) property's <see cref="System.Type"/>.
+        /// </param>
+        /// <param name="isStringProperty">
```

Comment:
Previous code was a bit odd, using string comparisons on type names.  This is more explicit and allows us, if we wish, to handle implicit conversions from strings.